proto/internal/temporal/server/api/taskqueue/v1/message.proto

```diff
@@ -80,16 +80,18 @@ message BuildIdRedirectInfo {
     string assigned_build_id = 1;
 }
 
-// Information about task forwarding from one partition to its parent. Versioning decisions are made
-// at the source partition and sent to the parent partition in this message.
+// Information about task forwarding from one partition to its parent. Versioning decisions for activity/workflow
+// tasks are made at the source partition and sent to the parent partition in this message so that parent partition
+// does not have to make versioning decision again. For Query/Nexus tasks, this works differently as the child's
+// versioning decision is ignored and the parent partition makes a fresh decision.
```

Comment:
I'm confused -- the description of the PR and our discussion on Wednesday made me think that we _should_ re-evaluate versioning decisions for activity/workflow tasks at the parent partition

Reply:
Just postponing that until overall versioning design is more clear. Because it was a bigger change, I did not want to spend a lot of time on it just right now.

Reply:
does this PR still fix the flakiness in this test then? https://github.com/temporalio/sdk-go/blob/master/test/worker_versioning_test.go#L885 

AFAIK the test is only about versioning rules, not Nexus/Query. Does the Nexus/Query UserData re-eval change the test behavior?

Reply:
Let's just make sure that the PR title reflects the change being made -- do you think it would be more clear if it said "Evaluate versioning rules in parent partition for forwarded Queries and Nexus tasks" ?

Reply:
Yes it fixes the test because the test is querying the wf and it was failing because the wf query was timing out.